src/rateEngine/BillingDeterminantsFactory.ts

```diff
@@ -2,6 +2,7 @@ import AnnualDemand from './billingDeterminants/AnnualDemand';
 import BlockedTiersInDays from './billingDeterminants/BlockedTiersInDays';
 import BlockedTiersInMonths from './billingDeterminants/BlockedTiersInMonths';
 import DemandPerDay from './billingDeterminants/DemandPerDay';
+import DemandPerDayAveraged from './billingDeterminants/DemandPerDayAveraged';
```

Comment:
Our naming conventions are complicated (in the literal sense of the word of mashing multiple things together) because we're encoding the "calculation period" in them along with other types of "periods"... in this case something like "averaging period".

We might consider splitting those out. So:

* `AveragedDemand` as the RateElement `type`
* `calculationPeriod` as a billing determinant option (`monthly` in our specific use case)
* `averagingPeriod` as a billing determinant option (`daily` in our specific use case)
* `qtyAveragingPeriods` as a billing determinant option (`3` in our specific use case)

It's a bit more implementation work, and it's solving for use cases that don't actually exist. So the real upside is just clearer naming for Future Us.

What do you think?

Reply:
@bknoles  I agree clearer naming would be a plus, but I do question if splitting out some of the conceptual options into arguments is unnecessary future-proofing, just because this way of calculating bills seems so specific to this one use case.

With your suggestion, are you implying that we should have multiple options implemented for, say, `calculationPeriod`, such as `monthly`, `weekly`, `yearly`, or that the current `monthly` implementation suffices but we still take it as an argument to show Future Us that the bill _could_ be calculated differently based on that option?

Reply:
For posterity, here's a sketched out version of a `Demand` strategy that we talked about internally which introduces a `DemandProfile` which knows how to handle _all_ of the ways that demand rates vary: tiers, TOU, different period types, and averaging as well. It would replace all of the demand strategies we currently have.

https://gist.github.com/bknoles/bee18a12804c0477d95b9504e9183903